### DISCUSSION TITLE
How to set a timeout to a KubernetesPodOperator task?

### DISCUSSION THREAD

**Discussion by davidbelhamou** [apache/airflow#26840] [Q&A] [answered]
Hi,

I'm using papermill to run sone notebooks, and i'm doing it using KubernetesPodOperator.
the operator create each time a pod that run a specific notebook.
In one of my notebook i had an error and the dag was still running after more than 24 hours.
How can i set a TTL to the KubernetesPodOperator? if there is no such option may someone have an idea?
Thanks in advance

**Comment by davidbelhamou** [comment, marked as answer]
Look like there is no TTL.
I created one using thread.

**Reply by bd-jahn** [reply]
Hi, do you have any update on this?

**Comment by Adaverse** [comment]
You can add `.spec.activeDeadlineSeconds` to your pod template or you can also add a timeout on the task itself called `execution_timeout`
